queries.sql

```diff
@@ -1,60 +1,204 @@
-SELECT name,full_name FROM animals RIGHT JOIN owners ON animals.owner_id=owners.id;
-SELECT species.name,COUNT(*) FROM animals JOIN species ON animals.species_id=species.id GROUP BY species.id;
-SELECT animals.name FROM animals JOIN owners ON animals.owner_id=owners.id JOIN species ON animals.species_id=species.id WHERE owner_id=2 AND species_id=2;
-SELECT name FROM animals JOIN owners ON animals.owner_id=owners.id WHERE owner_id=5 AND escape_attempts=0;
-SELECT full_name,COUNT(owner_id) FROM animals JOIN owners ON animals.owner_id=owners.id GROUP BY full_name ORDER BY COUNT(owner_id) DESC LIMIT 1;
+CREATE INDEX idx_animals_id ON visits (animals_id);
+CREATE INDEX idx_vets_id ON visits (vets_id);
+CREATE INDEX idx_email ON owners (email);
 
-SELECT animals.name AS last_seen_animal FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id WHERE vets.id=1 ORDER BY date_of_visit DESC LIMIT 1;
-SELECT COUNT(animals.name) FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id WHERE vets.id=3 GROUP BY vets.name;
-SELECT vets.name AS Vets,species.name AS Specialities FROM specializations JOIN species ON specializations.species_id=species.id RIGHT JOIN vets ON specializations.vets_id=vets.id;
-SELECT animals.name,date_of_visit FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id WHERE vets.id=3 AND date_of_visit BETWEEN '2020-04-01' AND '2020-08-30';
-SELECT animals.name,COUNT(animals.name) FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id GROUP BY animals.name ORDER BY COUNT(animals.name) DESC LIMIT 1;
-SELECT animals.name,date_of_visit FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id WHERE vets.id=2 ORDER BY date_of_visit ASC LIMIT 1;
-SELECT animals.name,date_of_birth,escape_attempts,neutered,weight_kg,vets.name AS vets_name,vets.age AS vets_age,vets.date_of_graduation AS vets_graduation,date_of_visit FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id ORDER BY date_of_visit DESC LIMIT 1;
-SELECT COUNT(*) FROM visits JOIN vets ON visits.vets_id=vets.id JOIN animals ON visits.animals_id=animals.id JOIN species ON animals.species_id=species.id LEFT JOIN specializations ON vets.id=specializations.vets_id AND species.id=specializations.species_id WHERE specializations.id IS NULL;
-SELECT species.name, COUNT(species.name) AS speciality FROM vets JOIN visits ON vets.id=visits.vets_id JOIN animals ON visits.animals_id=animals.id JOIN species ON animals.species_id=species.id WHERE vets.name='Maisy Smith' GROUP BY species.name ORDER BY speciality DESC LIMIT 1;
+EXPLAIN ANALYZE SELECT COUNT(*) FROM visits where animal_id = 4;
+EXPLAIN ANALYZE SELECT * FROM visits where vet_id = 2;
+EXPLAIN ANALYZE SELECT * FROM owners where email = 'owner_18327@mail.com';
```

Comment:
👍 Great effort in optimizing your SQL queries! Optimizing queries can be challenging, but you're on the right track. Let's delve into the details:

- [x] Your first query, concerning `animals_id`, appears to be working well. Both before and after creating the index, it returns the same number of rows (10 rows), which is a positive sign. 👍 

- [ ] However, in the second query, focusing on `vet_id`, there's a slight inconsistency. Before creating the index, it shows **5 rows**, but after indexing, it shows **6 rows**. Additionally, it seems you've applied a query limit after indexing. To gain a clearer understanding, I recommend running both queries in the same database, ensuring they operate on the same dataset, and executing them without any query limits. Please provide screenshots of the results for both scenarios.

| Before indexing | After indexing|
| --- | --- |
|  ![image](https://github.com/malikhaiderkhan/Vet-Clinic/assets/55840999/99ddc327-71fc-401d-9077-7b1ed81a71eb)| ![image](https://github.com/malikhaiderkhan/Vet-Clinic/assets/55840999/099ba8c3-9dd3-44d9-b0b5-dd74a078b6a9)|

- [ ] Regarding the third query, even though you haven't applied a query limit, there's a discrepancy in the results. Before indexing, it shows 8 rows, but after indexing, it returns 4 rows. To investigate this further, I suggest running both queries in the same database, containing an identical dataset, and sharing screenshots of the results for both queries.

| Before indexing | After indexing|
| --- | --- |
|  ![image](https://github.com/malikhaiderkhan/Vet-Clinic/assets/55840999/f335df7b-cb62-4371-a69b-b57833aa08cd)| ![image](https://github.com/malikhaiderkhan/Vet-Clinic/assets/55840999/06eedb4d-e7ae-470d-8030-45f96d5261a9) |

By performing these tests and sharing the results, we can better understand the impact of indexing on your queries. Keep up the excellent work in optimizing your database operations! 👍🔍📊